v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -117,27 +122,60 @@ export async function* testReporter(
     for (const [index, failure] of failures.entries()) {
       yield `\n${chalk.bold(chalk.red(`Failure (${index + 1})`))}: ${failure.name}\n`;
 
-      if (failure.decodedLogs !== undefined && failure.decodedLogs.length > 0) {
-        yield `Decoded Logs${chalk.grey(`: ${failure.decodedLogs.join("\n")}\n`)}`;
+      const stackTrace = failure.stackTrace();
+      if (
+        stackTrace !== undefined &&
+        stackTrace !== null &&
+        stackTrace.length > 0
+      ) {
+        const stackTraceMessage = getMessageFromLastStackTraceEntry(
+          stackTrace[stackTrace.length - 1],
+        );
+
+        const stackTraceStack: string[] = [];
+        for (const entry of stackTrace.reverse()) {
+          const callsite = encodeStackTraceEntry(entry);
+          if (callsite !== undefined) {
+            stackTraceStack.push(`  at ${callsite.toString()}`);
+          }
+        }
+
+        if (stackTraceMessage !== undefined || stackTraceStack.length > 0) {
+          yield `${stackTraceMessage ?? "Stack trace:"}\n${chalk.grey(stackTraceStack.join("\n"))}\n`;
+        }
+      }
+
+      if (
+        failure.decodedLogs !== undefined &&
+        failure.decodedLogs !== null &&
+        failure.decodedLogs.length > 0
+      ) {
+        yield `Decoded Logs:\n${chalk.grey(failure.decodedLogs.map((log) => `  ${log}`).join("\n"))}\n`;
```

Comment:
I added an extra newline before printing out the logs here to make it consistent with how stack traces are displayed. It should make the logs more readable. See the example output in the description.